index.js

```diff
@@ -8,20 +8,21 @@
 'use strict'
 
 const prepack = require('prepack')
-const utils = require('rollup-pluginutils')
 
 module.exports = function rollupPluginPrepack (options) {
-  options = Object.assign({ include: '**/*.js' }, options)
-
-  const filter = utils.createFilter(options.include, options.exclude)
-  const handle = (str) => `export default ${JSON.stringify(str.trim())}`
+  const handle = (str) => `export default ${str.trim()}`
 
   return {
     name: 'prepack',
-    transform: (source, id) => {
-      if (!filter(id)) return null
+    transform: (fileContents, filePath) => {
```

Comment:
You repo is using `transformBundle`, isn't that better? i'm totally agree on changing to it, it make sense.

Reply:
also i think that the second argument is options object? Sorry, i'm may be a bit out of that if that was changed in recent Rollup versions.

Reply:
> You repo is using transformBundle, isn't that better? i'm totally agree on changing to it, it make sense.

No, check out https://github.com/rollup/rollup/wiki/Plugins#creating-plugins

Reply:
> also i think that the second argument is options object? Sorry, i'm may be a bit out of that if that was changed in recent Rollup versions.

The second argument is the `filePath` string basically.

Reply:
Ah yea, really, it's typically called `id` which is the full filepath, yea.

The thing that we can do is to add the `options() {}` to the return, or it was something similar, i'll check here and there.

Reply:
Yea, as in here `https://github.com/rollup/rollup-plugin-node-resolve/blob/master/src/index.js#L72-L74`

`options() {}` method which gets the options. Probably something along these lines

```js
return {
  options(opts) {
    inputFile = opts.input.file
  }
}
```

Reply:
Oh, no, forget. Your approach in the repo make total sense. Change it as there - plugin accepts the dist file as 1st param, and options as 2nd param.

Reply:
Exactly!
In that case we can use `transformBundle` in the same way as Uglify does it https://github.com/TrySound/rollup-plugin-uglify/blob/master/index.js#L10

Reply:
note that in this case I have used `transform` to pass not the filename in the plugins array:

```js
{
  plugins: [
    typescript(),
    prepack({ only options here }),
    uglify(),
    // etc
  ]
}
```

Reply:
or do you prefer other approach ?

Reply:
Yea, but is Prepack need filepath or not? Is it required, if so, then we can just allow passing options as first param. Otherwise options should be second.

Reply:
Here my tests:

__1. Using transformBundle:__

`transformBundle` is an alias of `transformChunk(source, outputOptions, chunk)`:
> (formerly transformBundle) chunk transformer function called for each Rollup output chunk file. See https://github.com/rollup/rollup/wiki/Plugins

Received input:

```js
'use strict';

['A', 'B', 42].forEach(function (x) {
  var name = '_' + x.toString()[0].toLowerCase();
});
```

Prepack output:
```js
"use strict";
```

__2. Using transform:__

`transform( source, id )`:
> module transformer function. See https://github.com/rollup/rollup/wiki/Plugins

Received input:
```js
var self = {}
;['A', 'B', 42].forEach(function (x) {
  var name = '_' + x.toString()[0].toLowerCase()
  var y = parseInt(x)
  self[name] = y || x
})
```

Prepack output:
```js
(function () {
  var _$0 = this;

  var _0 = {
    _a: "A",
    _b: "B",
    _4: 42
  };
  _$0.self = _0;
}).call(this);
```

Conclusion: In this case we need to use `transform` function because it's applied to the module rather than chunk.